src/dataflow/src/source/s3.rs

```diff
@@ -603,47 +602,35 @@ enum DownloadError {
     SendFailed,
 }
 
-#[derive(Debug)]
+#[derive(Debug, thiserror::Error)]
 enum S3Error {
+    #[error("Unable to get S3 object {}/{}: {}", bucket, key, err)]
     GetObjectError {
         bucket: String,
         key: String,
+        #[source]
         err: SdkError<GetObjectError>,
     },
+    #[error("Unable to list S3 object {}: {}", bucket, err)]
     ListObjectsFailed {
         bucket: String,
+        #[source]
         err: SdkError<ListObjectsV2Error>,
     },
+    #[error("IO Error for S3 bucket {}: {}", bucket, err)]
     IoError {
         bucket: String,
+        #[source]
         err: std::io::Error,
     },
 }
 
-impl std::error::Error for S3Error {}
```

Comment:
note the missing `source` impl here for the the underlying `Sdk` AND `io` errors